server/tilesource/geospatial.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#############################################################################
+
+import gdal
+import osr
+import mapnik
+from PIL import Image
+import math
+
+from .base import FileTileSource
+
+try:
+    import girder
+    from girder import logger
+    from .base import GirderTileSource
+except ImportError:
+    girder = None
+    import logging as logger
+    logger.getLogger().setLevel(logger.INFO)
+
+
+class MapnikTileSource(FileTileSource):
+    name = 'mapnik'
```

Comment:
In keeping with other tile sources, change the non-girder name to `'mapnikfile'`.

Should the entire tile source be called `mapnik` rather than `geospatial`?

Also, add `cacheName = 'tilesource'`, and wrap the class with `@six.add_metaclass(LruCacheMetaclass)` so that we cache the tile source.  This will avoid calling `gdal.Open` for each tile.